lib/chef/provider/dsc_resource.rb

```diff
@@ -0,0 +1,145 @@
+#
+# Author:: Adam Edwards (<adamed@getchef.com>)
+#
+# Copyright:: 2014, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/util/powershell/cmdlet'
+require 'chef/util/dsc/local_configuration_manager'
+require 'chef/mixin/powershell_type_coercions'
+require 'chef/util/dsc/resource_store'
+
+class Chef
+  class Provider
+    class DscResource < Chef::Provider
+      include Chef::Mixin::PowershellTypeCoercions
+
+      provides :dsc_resource, os: "windows"
+
+      def initialize(new_resource, run_context)
+        super
+        @new_resource = new_resource
+        @resource_converged = false
+        @module_name = new_resource.module_name
+      end
+
+      def action_run
+        if ! @resource_converged
+          converge_by(generate_description) do
+            result = set_resource
+          end
+        end
+      end
+
+      def load_current_resource
+        @resource_converged = test_resource
+      end
+
+      def whyrun_supported?
+        true
+      end
+
+      def define_resource_requirements
+        requirements.assert(:run) do |a|
+          a.assertion { supports_dsc_invoke_resource? }
+          err = "Assuming a previous resource installs Powershell 5.0.10018.0 or higher."
```

Comment:
This is a confusing message to get in really-run mode. I think this string needs to get split up like in the next assertion.